src/platform/win/main.cpp

```diff
@@ -636,21 +641,25 @@ mat4 convToMat4(const vr::HmdMatrix34_t &m) {
                 m.m[0][3], m.m[1][3], m.m[2][3], 1.0f);
 }
 
-void vrUpdateInput() {
+void vrUpdateInput() { // going to use action manifest and ivr:input(Steam Vr Input)
     if (!hmd) return;
     vr::VREvent_t event;
-
+	char buffer[1024] = "test";
     while (hmd->PollNextEvent(&event, sizeof(event))) {
-        //ProcessVREvent( event );
+        //ProcessVREvent( event ); // eventually going to have the switch
         switch (event.eventType) {
             case vr::VREvent_TrackedDeviceActivated:
                 //SetupRenderModelForTrackedDevice( event.trackedDeviceIndex );
+				vr::RenderModel_t ** controllerRender;
+				hmd->GetStringTrackedDeviceProperty(event.trackedDeviceIndex, vr::ETrackedDeviceProperty::Prop_RenderModelName_String, buffer, 1024); // can be filled with an error,but I can't find the right type
+				rm->LoadRenderModel_Async(buffer, controllerRender);
+				// need to process render model?
                 LOG( "Device %u attached. Setting up render model\n", event.trackedDeviceIndex);
                 break;
             case vr::VREvent_TrackedDeviceDeactivated:
                 LOG("Device %u detached.\n", event.trackedDeviceIndex);
```

Comment:
perhaps this is a good place to call Input::reset() if we lost controller device